cudaaligner/src/hirschberg_myers_gpu.cu

```diff
@@ -312,44 +314,51 @@ myers_compute_scores(
         {
             if (threadIdx.x == 0)
                 score(0, reverse ? 1 : 0) = query_size;
-            __syncwarp();
         }
     }
 
+    const int32_t n_warp_iterations = ceiling_divide(n_words, warp_size) * warp_size;
+    __syncwarp();
     for (int32_t t = 1; t <= target_size; ++t)
     {
         int32_t warp_carry = 0;
         if (threadIdx.x == 0)
             warp_carry = 1; // for global alignment the (implicit) first row has to be 0,1,2,3,... -> carry 1
-        for (int32_t idx = threadIdx.x; idx < n_words; idx += warp_size)
+        for (int32_t idx = threadIdx.x; idx < n_warp_iterations; idx += warp_size)
         {
-            const uint32_t warp_mask = idx / warp_size < n_words / warp_size ? 0xffff'ffffu : (1u << (n_words % warp_size)) - 1;
+            if (idx < n_words)
+            {
+                const uint32_t warp_mask = idx / warp_size < n_words / warp_size ? 0xffff'ffffu : (1u << (n_words % warp_size)) - 1;
```

Comment:
pedantic comment - safer to put `(...)` around `(idx / warp_size)` and `(n_words / warp_size)`?

Reply:
To me operator precedence of `/` vs. `<` seems rather obvious - it's like in math.
Unless you feel strongly about it, I would like to leave it as it is.

Reply:
Nah don't feel strongly enough. Just a personal preference. Resolving